File: web/src/data/pro-networx-pricing.ts
/**
 * Pro Networks subscription pricing.
 * Single source of truth on the web side; mobile mirrors these values in
 * mobile/lib/core/constants/pro_networx_pricing.dart.
 *
 * The Stripe coupon (duration: once, amount_off: 500) handles the actual
 * billing; these constants only drive display.
 */

export const PRO_NETWORX_REGULAR_CENTS = 999;
export const PRO_NETWORX_INTRO_CENTS = 499;

export function formatProNetworxPriceUsd(cents: number): string {
  return `$${(cents / 100).toFixed(2)}`;
}

export const PRO_NETWORX_REGULAR_DISPLAY = formatProNetworxPriceUsd(
  PRO_NETWORX_REGULAR_CENTS,
);
export const PRO_NETWORX_INTRO_DISPLAY = formatProNetworxPriceUsd(
  PRO_NETWORX_INTRO_CENTS,
);

File: mobile/lib/core/constants/pro_networx_pricing.dart
// Pro Networks subscription pricing.
// Mirrors web/src/data/pro-networx-pricing.ts so display stays in sync.
// The Stripe coupon (duration: once, amount_off: 500) handles the actual
// billing; these constants only drive display.

const int proNetworxRegularCents = 999;
const int proNetworxIntroCents = 499;

String formatProNetworxPriceUsd(int cents) =>
    '\$${(cents / 100).toStringAsFixed(2)}';

final String proNetworxRegularDisplay =
    formatProNetworxPriceUsd(proNetworxRegularCents);
final String proNetworxIntroDisplay =
    formatProNetworxPriceUsd(proNetworxIntroCents);

/// Soft promo shown while messaging is free during beta.
final String proNetworxMessagingBetaPromo =
    'Messaging is free during beta. Pro-Networx is $proNetworxIntroDisplay '
    'first month, then $proNetworxRegularDisplay/mo — or Pro Bundle '
    '(Radio + Networx) for \$12.99/mo.';
